src/Data/Textile/Material/Origin.elm

```diff
@@ -2,50 +2,108 @@ module Data.Textile.Material.Origin exposing
     ( Origin(..)
     , decode
     , threadProcess
+    , toLabel
+    , toMicrofibersComplement
     , toString
     )
 
+import Data.Unit as Unit
 import Json.Decode as Decode exposing (Decoder)
+import Json.Decode.Extra as DE
 
 
 type Origin
-    = Natural
-    | Artificial
+    = ArtificialFromInorganic
+    | ArtificialFromOrganic
+    | NaturalFromAnimal
+    | NaturalFromVegetal
     | Synthetic
 
 
 decode : Decoder Origin
 decode =
-    Decode.map fromString Decode.string
+    Decode.string
+        |> Decode.andThen (fromString >> DE.fromResult)
 
 
-fromString : String -> Origin
+fromString : String -> Result String Origin
 fromString origin =
     case origin of
-        "Naturelles" ->
-            Natural
+        "ArtificialFromInorganic" ->
+            Ok ArtificialFromInorganic
 
-        "Artificielles" ->
-            Artificial
+        "ArtificialFromOrganic" ->
+            Ok ArtificialFromOrganic
 
-        "Synthétiques" ->
-            Synthetic
+        "NaturalFromAnimal" ->
+            Ok NaturalFromAnimal
+
+        "NaturalFromVegetal" ->
+            Ok NaturalFromVegetal
+
+        "Synthetic" ->
+            Ok Synthetic
 
         _ ->
-            Natural
+            Err <| "Origine inconnue: " ++ origin
+
+
+toMicrofibersComplement : Origin -> Unit.Impact
+toMicrofibersComplement origin =
+    -- Note: a malus expressed as a negative µPts/kg impact
```

Comment:
Would you link to point to the formula in the docs? Also maybe explain that it's returning `Ref*1000`?

Reply:
e6765db9